pkg/clusteragent/admission/mutate/config/config.go

```diff
@@ -221,46 +221,25 @@ func injectionMode(pod *corev1.Pod, globalMode string) string {
 // injectExternalDataEnvVar injects the External Data environment variable.
 // The format is: it-<init>,cn-<container_name>,pu-<pod_uid>
 func injectExternalDataEnvVar(pod *corev1.Pod) (injected bool) {
-	type containerInjection struct {
-		container *corev1.Container
-		init      bool
-	}
-	var containerInjections []containerInjection
-
-	// Collect all containers and init containers
-	for i := range pod.Spec.Containers {
-		containerInjections = append(containerInjections, containerInjection{&pod.Spec.Containers[i], false})
-	}
-	for i := range pod.Spec.InitContainers {
-		containerInjections = append(containerInjections, containerInjection{&pod.Spec.InitContainers[i], true})
-	}
-
-	// Inject External Data Environment Variable for each container
-	for _, containerInjection := range containerInjections {
-		if containerInjection.container == nil {
-			_ = log.Errorf("Cannot inject identity into nil container")
-			continue
-		}
-
-		containerInjection.container.Env = append([]corev1.EnvVar{
-			{
-				Name: podUIDEnvVarName,
-				ValueFrom: &corev1.EnvVarSource{
-					FieldRef: &corev1.ObjectFieldSelector{
-						FieldPath: "metadata.uid",
-					},
-				},
-			},
-			{
-				Name:  ddExternalDataEnvVarName,
-				Value: fmt.Sprintf("%s%t,%s%s,%s$(%s)", externalDataInitPrefix, containerInjection.init, externalDataContainerNamePrefix, containerInjection.container.Name, externalDataPodUIDPrefix, podUIDEnvVarName),
+	// Inject Internal Pod UID
+	injected = common.InjectEnv(pod, corev1.EnvVar{
+		Name: podUIDEnvVarName,
+		ValueFrom: &corev1.EnvVarSource{
+			FieldRef: &corev1.ObjectFieldSelector{
+				FieldPath: "metadata.uid",
 			},
-		}, containerInjection.container.Env...)
+		},
+	})
 
-		injected = true
-	}
+	// Inject External Data Environment Variable for the pod
+	injected = common.InjectDynamicEnv(pod, func(container *corev1.Container, init bool) (corev1.EnvVar, error) {
```

Comment:
for clarity you can create the function outside the `InjectDynamicEnv` call

Reply:
Agreed 👍